src/common/config_opts.h

```diff
@@ -16,6 +16,7 @@
 OPTION(host, OPT_STR, "") // "" means that ceph will use short hostname
 OPTION(fsid, OPT_UUID, uuid_d())
 OPTION(public_addr, OPT_ADDR, entity_addr_t())
+OPTION(public_bind_addr, OPT_ADDR, entity_addr_t())
```

Comment:
we could make this "mon_public_bind_addr" to be more specific. the same behavior could be carried over to OSD, MDS etc. if we wanted to so I left a bit more general. Note that we could also add "cluster_bind_addr" etc. if we wanted to too.